sdcm/nemesis.py

```diff
@@ -1661,6 +1662,63 @@ def disrupt_nodetool_enospc(self, sleep_time=30, all_nodes=False):
                     finally:
                         clean_enospc_on_node(target_node=node, sleep_time=sleep_time)
 
+    def disrupt_end_of_quota_nemesis(self, sleep_time=30, all_nodes=False):
+        """
+        Nemesis flow
+        ---------------------
+        1. Check quota is enabled in the SCT configuration file.
+        2. Verify quota is enabled on the node.
+        3. Check disk usage.
+        4. Define quota size same as disk usage + 1GB.
+        5. Wait for end of quota message to appear in the log - "Storage I/O error: 122: Disk quota exceeded"
+        6. Except I/O Error if happens.
+        7. Remove the quota by setting it to 0
+        8. Restart scylla server
+        9. Verify scylla is up
+        """
+        if not self.cluster.params.get("enable_quota"):
+            raise UnsupportedNemesis("Quota configuration is not enabled on the cluster thus skipping this Nemesis")
+
+        if all_nodes:
+            nodes = self.cluster.nodes
+            InfoEvent('EDQUOT test on {}'.format([n.name for n in nodes])).publish()
+        else:
+            nodes = [self.target_node]
+
+        for node in nodes:
+            if self._is_it_on_kubernetes():
+                LOGGER.info("Skipping nemesis for kubernetes")
+                break
+            node.verify_quota_is_enabled_on_node()
```

Comment:
`verify_quota_is_configured_for_scylla_user` and `configure_quota_on_node` use target node as argument, why here is different?

Reply:
Why we have methods in `BaseScyllaCluster` and `BaseNode` classes and other methods are in sdcm/utils/common.py? 
`verify_quota_is_enabled_on_node` is related to node setup and verify the possibility of adding quota configuration (enabled). So, I added it to `BaseNode` class.
Maybe I should change the name for `configure_quota_on_node` to `configure_quota_on_node_for_scylla_user`.
Anyway, both these methods (`verify_quota_is_configured_for_scylla_user` and `configure_quota_on_node`) are in the utils/common.py and require a parameter as target_node.
That's the logic of SCT which is not very clear to me why there is a division between `BaseNode` and methods in `common.py` and other files in `sdcm/utils`.